src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -141,20 +145,26 @@ public void setPort(final int port) {
   public void connect() {
     if (!isConnected()) {
       try {
-        socket = new Socket();
-        // ->@wjw_add
-        socket.setReuseAddress(true);
-        socket.setKeepAlive(true); // Will monitor the TCP connection is
-        // valid
-        socket.setTcpNoDelay(true); // Socket buffer Whetherclosed, to
-        // ensure timely delivery of data
-        socket.setSoLinger(true, 0); // Control calls close () method,
-        // the underlying socket is closed
-        // immediately
-        // <-@wjw_add
-
-        socket.connect(new InetSocketAddress(host, port), connectionTimeout);
-        socket.setSoTimeout(soTimeout);
+        File udsFile = new File(host);
+        if (udsFile.exists()) {
+          socket = AFUNIXSocket.newInstance();
```

Comment:
We're omitting all the socket options here. I'm not sure which apply specifically to TCP sockets but I'm sure that some of them `should` apply to UDS as well such as SoTimeout. Can we re-check this again plz?